lib/asset-rev.js

```diff
@@ -24,9 +25,19 @@ function AssetRev(inputTree, options) {
   this.ignore = options.ignore;
   this.description = options.description;
 
+  // first pass - excludes replaceable source code
+  this.exclude = exclude.slice();
+  this.replaceExtensions.forEach(function(ext) { this.exclude.push('**/*.' + ext); }, this);
```

Comment:
Thank you for the detailed explanation. I'd prefer to avoid the breaking change for now (as you have done here), so that we can fix the primary bug that this is addressing.